src/NBitcoin/Block.cs

```diff
@@ -116,7 +133,7 @@ public uint256 GetHash()
             if (hash != null)
                 return hash;
 
-            if (Block.BlockSignature)
+            if (this.GetNetworkOptions().IsProofOfStake)
```

Comment:
What if Options was passed in `GetHash(optiosn)` is it a big change?
Just asking dont act on this.

Reply:
See below.